src/linux/backend-v4l2.cpp

```diff
@@ -761,9 +766,22 @@ namespace librealsense
 
             int max_fd = std::max(std::max(_stop_pipe_fd[0], _stop_pipe_fd[1]), _fd);
 
-            struct timeval tv = {5,0};
-
-            auto val = select(max_fd+1, &fds, NULL, NULL, &tv);
+            struct timespec mono_time;
+            int r = clock_gettime(CLOCK_MONOTONIC, &mono_time);
```

Comment:
Probably marginal, but the return value should be checked as well for errno code.